openml-r-common/src/main/java/com/feedzai/openml/r/ClassificationGenericRModel.java

```diff
@@ -148,26 +152,21 @@ public synchronized void close() {
     }
 
     /**
-     * Uses the connection to Rserve to execute a expression. This method needs to be synchronized because the
-     * connection to Rserve isn't thread-safe.
+     * Calls a R {@code function} to evaluate an {@code instance}.
+     * This function depends on the {@link RConnection connection to Rserve} to run R code. The connection to Rserve
+     * isn't thread-safe and so this method needs to be synchronized.
```

Comment:
since we are on this topic: can you add a link on this class javadoc where it says/explains that the rconnection isn't thread safe? 

While it is not a 1st level topic, I think this link referes to this topic: http://rforge.net/Rserve/

> Rserve is thread safe across connections, but eval methods are not thread safe within one connection. This means that multiple threads should not use the same connection unless they guarantee that no eval calls are run in parallel.

Reply:
ping

Reply:
I added a link at the end of the javadoc
`@see <a href="http://rforge.net/Rserve/">Rserve</a>`

I searched their source code but the only reference to a thread-safe problem doesn't explain anything.
https://github.com/s-u/Rserve/blob/master/src/include/sbthread.h
> /* this isn't really thread-safe, but creating more mutexes
     at exactly the same time by different threads on an SMP machine
     is ... let's say inprobable. moreover ++ might be atomic.
     creating yet another mutex for this single operation seems
     to me as an overkill. but if you need 100% thread-safe code,
     feel free to implement it here ;) */

I prefer the description presented on their website (http://rforge.net/Rserve/)
> Rserve is thread safe across connections, but eval methods are not thread safe within one connection. This means that multiple threads should not use the same connection unless they guarantee that no eval calls are run in parallel.